src/engine.rs

```diff
@@ -80,10 +81,42 @@ impl Engine {
             PipelineData::Empty,
         )
         .and_then(|v| v.into_value(Span::unknown()))
-        .map(|v| RunCodeResult::Success(v.into()))
+        .map(|v| -> crate::types::Value {
+            v.try_into()
+                .unwrap_or_else(|v| crate::types::Value::html(self.value_to_html(v)))
+        })
+        .map(RunCodeResult::Success)
         .unwrap_or_else(|e| RunCodeResult::Error(e.into()))
     }
 
+    fn value_to_html(&mut self, value: Value) -> String {
+        let (block, working_set) = self.parse("to html -d --partial");
```

Comment:
- Refactor the code to a const and use it in the error messages
-  is there a long version to -d?